File: bench/takfp/takfp.cs
/* The Great Computer Language Shootout 
   http://shootout.alioth.debian.org/

   contributed by Isaac Gouy 
*/

using System;

class Takfp
{
   public static float Tak(float x, float y, float z) {
      if (y >= x) return z;
      return Tak(Tak(x-1.0f,y,z), Tak(y-1.0f,z,x), Tak(z-1.0f,x,y)); 
   }

   static void Main(string[] args){
      int n = 1;
      if (args.Length > 0) n = Int32.Parse(args[0]);
      Console.WriteLine("{0}", Tak(n*3.0f, n*2.0f, n*1.0f));
   }
}

File: bench/takfp/takfp.java
// $Id: takfp.java,v 1.2 2004-11-27 03:42:24 bfulgham Exp $
// http://shootout.alioth.debian.org/
// Contributed by Isaac Gouy

public class takfp {
    public static void main(String args[]) {
	int n = Integer.parseInt(args[0]);
	System.out.println( Tak(n*3.0d, n*2.0d, n*1.0d) );
    }

  public static double Tak (double x, double y, double z) {
    if (y >= x) return z;
    else return Tak(Tak(x-1,y,z), Tak(y-1,z,x), Tak(z-1,x,y)); 
  }
}
